### PR TITLE
d1: foreign keys docs

### PR DESCRIPTION
**Pull request by elithrar**
This PR adds:

- [x] New Foreign Key documentation covering how FKs are enforced and how to defer enforcement
- [x] How to use FK actions to cascade (or not) deletes/updates
- [x] Cross-link from both Import Data and Migrations guides, as these are common cases where relaxing FK constraints is important.
- [x] Moves important content into “Build with D1” - too much is buried under “Reference” and feedback is consistent about this being a catch-all

### CODE REVIEW

File: content/d1/build-with-d1/foreign-keys.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    name TEXT,
+    metadata TEXT
+)
+
+CREATE TABLE orders (
+    order_id INTEGER PRIMARY KEY,
+    status INTEGER,
+    item_desc TEXT,
+    shipped_date INTEGER,
+    FOREIGN KEY(user_who_ordered) REFERENCES users(user_id) ON DELETE RESTRICT
```
Comment (vy-ton) [inline content/d1/build-with-d1/foreign-keys.md]: For this syntax, have to define `user_who_ordered` column separately:

```
CREATE TABLE orders (
    order_id INTEGER PRIMARY KEY,
    status INTEGER,
    item_desc TEXT,
    shipped_date INTEGER,
    user_who_ordered INTEGER,
    FOREIGN KEY(user_who_ordered) REFERENCES Customers(CustomerId)
)

File: content/d1/build-with-d1/foreign-keys.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+To illustrate this with an example based on an e-commerce website with two tables:
+
+* A `users` table that defines common properties about a user account, including a unique `user_id` identifier
+* An `orders` table that maps an order back to a `user_id` in the user table.
+
+This mapping is defined as `FOREIGN KEY`, which ensures that:
+
+* You cannot delete a row from the `users` table that would violate the foreign key constraint. In practice, this means that you can't end up with orders that do not have a valid user to map back to.
+* `orders` are always defined against a valid `user_id`, mitigating the risk of creating orders that refer to invalid (or non-existant) users.
+* Deleting orders does not delete users. The `ON DELETE RESTRICT` action as part of the table definition enables this.
```
Comment (vy-ton) [inline content/d1/build-with-d1/foreign-keys.md]: Maybe delete `ON DELETE RESTRICT` from the example? From my reading, `ON DELETE RESTRICT` doesnt impact "deleting orders does not delete users" but only matters for multi-statement transactions and when FK constraint violation error is thrown.

File: content/d1/build-with-d1/foreign-keys.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    score_id INTEGER PRIMARY KEY,
+    game TEXT,
+    score INTEGER,
+    FOREIGN KEY(player_id) REFERENCES users(user_id) ON DELETE CASCADE
+)
+```
+
+## Relax foreign key constraints
+
+When running a [query](/d1/platform/client-api/), [migration](/d1/platform/migrations/) or [importing data](/d1/learning/importing-data/) against a D1 database, there may be situations in which you need to disable foreign key validation during table creation or changes to your schema.
```
Comment (vy-ton) [inline content/d1/build-with-d1/foreign-keys.md]: `/d1/platform/client-api/` -> `/d1/build-with-d1/d1-client-api/`

File: content/d1/build-with-d1/foreign-keys.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+* D1's foreign key enforcement is equivalent to SQLite's `PRAGMA foreign_keys = on` directive. Because D1 runs every query inside an implicit transaction, user queries cannot change this during a query or migration.
+* Instead, D1 allows you to call `PRAGMA defer_foreign_keys = true` or `false`, which allows you to violate foreign key constraints temporarily.
```
Comment (vy-ton) [inline content/d1/build-with-d1/foreign-keys.md]: Mention the scope of the PRAGMA statement and add PRAGMA statement to https://developers.cloudflare.com/d1/reference/database-commands/

File: content/d1/build-with-d1/foreign-keys.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+{{<Aside type="warning">}}
+
+If your migration or batch query does not resolve any outstanding foreign key violations, your migration or query will fail with an error.
```
Comment (vy-ton) [inline content/d1/build-with-d1/foreign-keys.md]: So there's no way to permanently have a table with a FK violation?

**Review by vy-ton — APPROVED** [review]
Preemptively approving once Matt addresses comments to unblock merge for Dev Week/D1 GA

File: content/d1/build-with-d1/foreign-keys.md (hunk 4 above)
Comment (elithrar) [reply content/d1/build-with-d1/foreign-keys.md]: great call. adding.

File: content/d1/build-with-d1/foreign-keys.md (hunk 5 above)
Comment (elithrar) [reply content/d1/build-with-d1/foreign-keys.md]: correct.

**Review by irvinebroque — APPROVED** [review]

**Review by GregBrimble — APPROVED** [review]

**Review by crwaters16 — APPROVED** [review]

**Pull request merged**